qiime2/sdk/usage.py

```diff
@@ -88,6 +88,14 @@ def build_opts(self, signature, scope):
                 if isinstance(self.values[name], ScopeRecord) \
                         and self.values[name].ref in scope.records:
                     value = self.values[name].result
+                elif (isinstance(self.values[name], list)
+                      and self.values[name]
+                      and isinstance(self.values[name][0], ScopeRecord)):
```

Comment:
Perhaps for a "belt-and-suspenders" approach here, we should consider checking that _all_ of the records in the collection are `ScopeRecord` instances. The same consideration applies to the `set` branch below.

Reply:
Would something like the following work to check the records in the collections mentioned above:

`all(isinstance(n, ScopeRecord) for n in self.values[name]))`

I've tested this solution for both set and list types that contain string values, and they both seem to work.

Reply:
Works for me!